base/strings/string.jl

```diff
@@ -118,7 +120,9 @@ function thisind(s::String, i::Int)
     return i
 end
 
-function nextind(s::String, i::Int)
+nextind(s::String, i::Int) = _nextind_str(s, i)
+
+function _nextind_str(s, i::Int)
```

Comment:
Maybe add `::AbstractString`?

Reply:
An optimal option would be to define `nextind(s::Union{String, SubString{String}}, i::Int)` (the same for `thisind`), but then I would have to move it to other place in the code after `SubString` is defined. What do you think?

Adding `::AbstractSting` is misleading as `_nextind_str` will not work for any `AbstractString` and it is intended for internal use only.

So I would be obliged for a guidance what is the optimal design of code for such cases (i.e. the same body of a method applies to two different types, but at the moment of definition of the method only one of those types is defined)?

Reply:
I'd say it's fine as it is given that bootstrapping is difficult. Maybe just add a comment saying that `s::Union{String, SubString{String}}`.

Reply:
Added - I hope this is what you meant.